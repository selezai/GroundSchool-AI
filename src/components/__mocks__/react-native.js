const React = require('react');

const View = ({ children, style, testID }) => 
  React.createElement('View', { style, testID }, children);

const Text = ({ children, style, testID }) => 
  React.createElement('Text', { style, testID }, children);

const TextInput = (props) => 
  React.createElement('TextInput', props);

const TouchableOpacity = ({ children, onPress, style, testID }) => 
  React.createElement('TouchableOpacity', { onPress, style, testID }, children);

const StyleSheet = {
  create: styles => styles,
};

module.exports = {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
};
