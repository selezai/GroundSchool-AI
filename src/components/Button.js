import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const Button = ({ 
  title, 
  onPress, 
  disabled = false,
  variant = 'primary', 
  size = 'medium',
  style, 
  textStyle,
  testID = 'custom-button' 
}) => {
  return (
    <TouchableOpacity 
      onPress={onPress} 
      disabled={disabled}
      style={[
        styles.button,
        styles[variant],
        styles[size],
        disabled && styles.disabled,
        style
      ]}
      testID={testID}
    >
      <Text 
        style={[
          styles.text, 
          styles[`${variant}Text`],
          disabled && styles.disabledText,
          textStyle
        ]} 
        testID="button-text"
      >
        {title}
      </Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(['primary', 'secondary', 'outline', 'danger']),
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  style: PropTypes.object,
  textStyle: PropTypes.object,
  testID: PropTypes.string
};

const styles = StyleSheet.create({
  button: {
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 8,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.2,
    shadowRadius: 1.5,
    elevation: 2,
  },
  primary: {
    backgroundColor: '#3182CE', // Aviation blue
    borderWidth: 0,
  },
  secondary: {
    backgroundColor: '#718096', // Cool gray
    borderWidth: 0,
  },
  outline: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: '#3182CE',
  },
  danger: {
    backgroundColor: '#E53E3E', // Red
    borderWidth: 0,
  },
  small: {
    paddingVertical: 6,
    paddingHorizontal: 12,
  },
  medium: {
    paddingVertical: 10,
    paddingHorizontal: 16,
  },
  large: {
    paddingVertical: 14,
    paddingHorizontal: 24,
  },
  disabled: {
    backgroundColor: '#E2E8F0',
    borderColor: '#CBD5E0',
    shadowOpacity: 0,
    elevation: 0,
  },
  text: {
    fontWeight: '600',
    textAlign: 'center',
  },
  primaryText: {
    color: '#FFFFFF',
  },
  secondaryText: {
    color: '#FFFFFF',
  },
  outlineText: {
    color: '#3182CE',
  },
  dangerText: {
    color: '#FFFFFF',
  },
  disabledText: {
    color: '#A0AEC0',
  }
});

export default Button;
