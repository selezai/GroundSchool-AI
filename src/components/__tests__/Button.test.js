import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import Button from '../Button';

describe('Button Component', () => {
  // Snapshot test
  it('renders correctly', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <Button title="Test" onPress={() => {}} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  // Props tests
  it('renders with different titles', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <Button title="Different Title" onPress={() => {}} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  // Interaction tests
  it('calls onPress when pressed', () => {
    const mockOnPress = jest.fn();
    let component;
    
    act(() => {
      component = renderer.create(
        <Button title="Press Me" onPress={mockOnPress} />
      );
    });

    const button = component.root.findByProps({ testID: 'custom-button' });
    act(() => {
      button.props.onPress();
    });

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  // Style tests
  it('applies custom styles', () => {
    const customStyle = { backgroundColor: 'red' };
    let tree;
    
    act(() => {
      tree = renderer.create(
        <Button 
          title="Styled Button" 
          onPress={() => {}} 
          style={customStyle}
        />
      ).toJSON();
    });

    expect(tree).toMatchSnapshot();
  });

  // Disabled state tests
  it('handles disabled state', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <Button 
          title="Disabled Button" 
          onPress={() => {}} 
          disabled={true}
        />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });
});

