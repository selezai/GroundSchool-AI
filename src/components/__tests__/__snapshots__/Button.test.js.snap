// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component applies custom styles 1`] = `null`;

exports[`Button Component handles disabled state 1`] = `null`;

exports[`Button Component renders correctly 1`] = `null`;

exports[`Button Component renders with different titles 1`] = `null`;
